askama_parser/src/node.rs

```diff
@@ -928,8 +928,23 @@ pub struct Extends<'a> {
 
 impl<'a> Extends<'a> {
     fn parse(i: &'a str) -> ParseResult<'a, Self> {
-        let (i, path) = preceded(ws(keyword("extends")), cut(ws(str_lit)))(i)?;
-        Ok((i, Self { path }))
+        let start = i;
+
+        let (i, (pws, _, (path, nws))) = tuple((
+            opt(Whitespace::parse),
+            ws(keyword("extends")),
+            cut(pair(ws(str_lit), opt(Whitespace::parse))),
+        ))(i)?;
+        if pws.is_some() || nws.is_some() {
+            Err(nom::Err::Failure(ErrorContext {
+                input: start,
+                message: Some(Cow::Borrowed(
+                    "whitespace control is not allowed on `extends`",
+                )),
+            }))
+        } else {
+            Ok((i, Self { path }))
+        }
```

Comment:
Nit: I'd prefer to either write this with an early return (probably for the `Ok` case) or just as a match:

```rust
match (pws, nws) {
     (None, None) => Ok((i, Self { path })),
     (_, _) => Err(..),
}
```

Reply:
:+1: